apps/cowswap-frontend/src/modules/hooksStore/dapps/PermitHookApp/index.tsx

```diff
@@ -0,0 +1,78 @@
+import { useCallback, useMemo, useState } from 'react'
+
+import { ButtonPrimary } from '@cowprotocol/ui'
+
+import { useGeneratePermitHook, usePermitInfo, useTokenSupportsPermit } from 'modules/permit'
+
+import { ContentWrapper, Row, Wrapper } from './styled'
+
+import { HookDappProps } from '../../types/hooks'
+import { TradeType } from 'modules/trade'
+import { Token } from '@uniswap/sdk-core'
+import { isAddress } from '@cowprotocol/common-utils'
+
+export function PermitHookApp({ isPreHook, context }: HookDappProps) {
+  const hookToEdit = context.hookToEdit
+  const [tokenAddress, setTokenAddress] = useState<string>()
+  const [spenderAddress, setSpenderAddress] = useState<string>()
+  const generatePermitHook = useGeneratePermitHook()
+  const token = useMemo(() => {
+    try {
+      return new Token(context.chainId, tokenAddress || '', 18)
```

Comment:
Now there's no amounts anywhere so this is not a problem, but might become one since many tokens have a different number of decimals.